apps/sms/services/js/moz_mobile_message/moz_mobile_message_shim.js

```diff
@@ -143,13 +168,22 @@ var MozMobileMessageShim = {
    */
   getThreads(stream) {
     var cursor = null;
+    var isStreamCancelled = false;
+
+    stream.cancel = (reason) => {
+      debug('getThreads stream is cancelled: %s', reason);
+      isStreamCancelled = true;
+      return Promise.resolve();
+    };
 
     // WORKAROUND for bug 958738. We can remove 'try\catch' block once this bug
     // is resolved
     try {
       cursor = mozMobileMessage.getThreads();
     } catch(e) {
-      console.error('Error occurred while retrieving threads: ' + e.name);
+      debug(
+        'Error occurred while retrieving threads: %s', e.message || e.name, e
+      );
```

Comment:
for real errors I'd like to keep console.error, so that they still show up in case debug is off.